### PR TITLE
Move http transport to httpclient5, move kinesis to separate module, update Flink integration's structure

### PR DESCRIPTION
**Pull request by mobuchowski**
To remove usage of vulnareble library `commons-codec` coming from outdated HTTP library https://github.com/OpenLineage/OpenLineage/issues/3182

- removed it's usage
- moved `KinesisTransport` to separate module, as underneath Kinesis depends on old http client
- restructured Flink integration - removed cyclic dependency between OL integration and example jobs, now the example jobs are build separately and depend on generated shadow jar, and they themselves produce shadow jar that's used in the integration tests, rather than having direct code dependency.

### CODE REVIEW

**Review by arturowczarek — APPROVED** [review]
The flink Gradle part is not completely clear to me, but apart from that everything looks good.

**Review by tnazarew — APPROVED** [review]
good stuff 👍

File: client/java/src/main/java/io/openlineage/client/transports/FileTransport.java [hunk 1]
```diff
@@ -46,11 +47,13 @@ public void emit(@NonNull OpenLineage.JobEvent jobEvent) {
 
   private void emit(String eventAsJson) {
     try {
-      FileUtils.writeStringToFile(
-          file,
-          eventAsJson.replace(System.lineSeparator(), "") + System.lineSeparator(),
-          StandardCharsets.UTF_8,
-          true);
+      Files.createDirectories(file.getParentFile().toPath());
```
[the remark this answers is not in the document]
Comment (mobuchowski) [reply client/java/src/main/java/io/openlineage/client/transports/FileTransport.java:50]: Removed usage of dependency, which was transitive - so no readding it as actual dependency.

**Comment by jtsolidatus** [comment]
Hello.

https://github.com/OpenLineage/OpenLineage/blob/2ff0ce731440213274cec77be97cb1fbd2248f70/client/java/transports-kinesis/src/main/resources/META-INF.services/io.openlineage.client.transports.TransportBuilder

The class this references does not exist.  This should be changed to point at :
io.openlineage.client.transports.kinesis.KinesisTransportBuilder

**Pull request merged**